event/src/reducers/API/tree.js

```diff
@@ -106,3 +87,72 @@ export default (state = initialState, action) => {
       return state;
   }
 };
+function mapNodesFromResult(action) {
+  if (action.dataUrl.toString().includes('/merge_requests/')) {
+    return filterData(action)
+      .map((node) => {
+        return {
+          name: node.new_path,
+          path: node.new_path,
+          isTree: false,
+          children: false,
+        };
+      })
+      .reduce((map, obj) => {
+        map[obj.name] = obj;
+        return map;
+      }, {});
+  } else {
+    return action.payload
+      .map((node) => {
+        return {
+          name: node.name,
+          path: node.path
+            .split("/")
+            .filter((pathSub) => pathSub.length !== 0),
+          isTree: node.type === "tree"
+            ? {
+              isOpen: false,
+            }
+            : false,
+          children: node.type === "tree" ? {} : undefined,
+        };
+      })
+      .reduce((map, obj) => {
+        map[obj.name] = obj;
+        return map;
+      }, {});
+  }
+
+};
+
+function filterData(action) {
+  let data = action.payload['changes'];
+  if (action.filtersEnabled['test']) {
+    data = data.filter((node) => {
+      return !node.new_path.includes('src/test/');
+    })
+  }
+  if (action.filtersEnabled['renamed']) {
+    data = data.filter((node) => {
+      return !node.renamed_file;
+    })
+  }
+  if (action.filtersEnabled['removed']) {
+    data = data.filter((node) => {
+      return !node.deleted_file;
+    })
+  }
+  if (action.filtersEnabled['newFile']) {
+    data = data.filter((node) => {
+      return !node.new_file;
+    })
+  }
+  if (action.filtersEnabled['imports']) {
+    data = data.filter((node) => {
+      return !node.new_file;
```

Comment:
functionaltiy not implemented yet: I wanted to filter all entries where the change was simple import changes. ie.:
- import a.b.c.d.EClass
+ import f.g.h.IClass

This information can be grepped by the rest call but processing it is most likely overkill.
That information is stored on node.diff as a string, it is a delta diff of the files and cumbersome to use.
I bet there are some libraries that could deal with that...

If you cannot make it work, scrap this one and remove the corresponding checkbox